flake.nix

```diff
@@ -182,62 +166,6 @@
           pkgs.lib.listToAttrs
         ];
 
-        apps.tests = let
```

Comment:
Please wrap the test script into app (or many). It's much easier to run `nix run .#tests.all` than to keep in memory the full `nix develop --command bash test-template.sh`.